### PR TITLE
MGMT-15346: Code refactor to support pkg/ignition unit-tests

### PR DESCRIPTION
**Pull request by nmagnezi**

### CODE REVIEW

**Comment by nmagnezi** [comment]
Tested okay:
```
INFO Reusing base CoreOS ISO from cache
INFO Reusing recovery CoreOS ISO from cache
INFO Successfully pulled container registry image
INFO Successfully pulled OpenShift 4.12.12 release images required for bootstrap
INFO Successfully pulled OpenShift 4.12.12 release images required for installation
INFO Successfully generated data ISO
INFO Successfully downloaded appliance base disk image
INFO Successfully extracted appliance base disk image
INFO Successfully generated appliance disk image
INFO Time elapsed: 7m21s
INFO
INFO Appliance disk image was successfully created in assets directory: assets/appliance.raw
```

File: pkg/ignition/ignition.go [hunk 1]
```diff
@@ -12,22 +12,48 @@ import (
 	"github.com/pkg/errors"
 )
 
+type OSInterface interface {
```
Comment (danielerez) [inline pkg/ignition/ignition.go:15]: Wasn't that added on another PR? Or is it just based on it:/

Comment (nmagnezi) [reply pkg/ignition/ignition.go:15]: that's a different pkg

Comment (nmagnezi) [reply pkg/ignition/ignition.go]: now using the generic fileutil.OSInterface

**Pull request merged**